main/src/ca/uwaterloo/flix/language/ast/Kind.scala

```diff
@@ -33,6 +33,25 @@ sealed trait Kind {
     */
   override def toString: String = Kind.ShowInstance.show(this)
 
+  /**
+    * The kind this kind extends.
+    */
+  protected val superKind: Option[Kind] = None
```

Comment:
I dont think we want this. The only place the sub-kinding will take place is deep inside the unification algorithm when a type variable is substituted for a type. Normally the check here would enforce the same kind, but we will now allow a bit more.

Reply:
https://github.com/flix/flix/blob/master/main/src/ca/uwaterloo/flix/language/phase/Unification.scala#L78

Reply:
The reason this check is not failing more at the moment is because type variables are not created with the right kind.